examples/transit_vnet_dedicated/main.tf

```diff
@@ -112,12 +112,13 @@ module "bootstrap" {
 module "outbound_bootstrap" {
   source = "../../modules/bootstrap"
 
-  # location               = var.location
-  resource_group_name      = azurerm_resource_group.this.name
-  create_storage_account   = false
-  existing_storage_account = module.bootstrap.storage_account.name
-  storage_share_name       = var.outbound_storage_share_name
-  files                    = var.outbound_files
+  resource_group_name    = azurerm_resource_group.this.name
+  create_storage_account = false
+  storage_account_name   = module.bootstrap.storage_account.name
+  storage_share_name     = var.outbound_storage_share_name
+  files                  = var.outbound_files
+
+  depends_on = [module.bootstrap]
```

Comment:
due to the way terraform resolves dependencies. Even though you use this:
```
existing_storage_account = module.bootstrap.storage_account.name
```
Terraform tries to parallelise execution of both modules, most probably  because the `output` is set to a `local` instead of a resource (regardless the fact that this local refers to a resource under some condition).

So there are situation, where the module that creates only a File Share runs before the actual storage is available. Explicit `depends_on` solves the problem.